pages/settings.js

```diff
@@ -0,0 +1,141 @@
+import { useEffect, useState } from "react";
+import {
+  Avatar,
+  Button,
+  Col,
+  DatePicker,
+  Divider,
+  Form,
+  Input,
+  Row,
+  Space,
+  Typography,
+  Upload,
+} from "antd";
+import moment from "moment";
+import { UploadOutlined } from "@ant-design/icons";
+import { useAuth, withAuth } from "~/components/Auth";
+import AppLayout from "~/components/layouts/AppLayout";
+import { getBase64 } from "~/lib/utils/images";
+
+const { Title } = Typography;
+
+const Section = ({ title }) => (
+  <Divider orientation="left">
+    <Title level={5} type="secondary">
+      {title}
+    </Title>
+  </Divider>
+);
+
+function Settings() {
+  const { user, edit_user, isLoading } = useAuth();
+  const [formPersonal] = Form.useForm();
+  const [formPassword] = Form.useForm();
+  const [avatar, setAvatar] = useState();
+
+  useEffect(() => {
+    setAvatar(user.photo);
+  }, [user]);
+
+  const breakpoints = {
+    xs: 24,
+    md: 12,
+    xl: 8,
+    xxl: 6,
+  };
+
+  return (
+    <AppLayout>
+      <Row justify="space-between">
+        <Col>
+          <Title level={2}>Configurações</Title>
+        </Col>
+        <Col>
+          <Space>
+            <Button
+              onClick={() => {
+                formPersonal.resetFields();
+                setAvatar(user.photo);
+              }}
+            >
+              Cancelar
+            </Button>
+            <Button
+              loading={isLoading}
+              onClick={() => {
+                formPersonal.submit();
+              }}
+              type="primary"
+            >
+              Guardar
+            </Button>
+          </Space>
+        </Col>
+      </Row>
+      <Form form={formPersonal} onFinish={edit_user} layout="vertical">
+        <Section title="Foto de Perfil" />
+        <Space>
+          <Avatar size={100} src={avatar} />
+          <Form.Item name="user[photo]">
+            <Upload
+              accept="image/*"
+              maxCount={1}
+              beforeUpload={(file) => {
+                getBase64(file, (imageUrl) => setAvatar(imageUrl));
+                return false;
+              }}
+              onRemove={() => setAvatar(user.photo)}
+            >
+              <Button icon={<UploadOutlined />}>Upload</Button>
+            </Upload>
+          </Form.Item>
+        </Space>
+        <Section title="Informações Pessoais" />
+        <Row gutter={24}>
+          <Col {...breakpoints}>
+            <Form.Item name="user[first_name]" label="Name">
+              <Input defaultValue={user.first_name} />
+            </Form.Item>
+          </Col>
+          <Col {...breakpoints}>
+            <Form.Item name="user[last_name]" label="Apelido">
+              <Input defaultValue={user.last_name} />
+            </Form.Item>
+          </Col>
+        </Row>
```

Comment:
I don't think you should be able to change your name, since kids will be using this and they shouldn't change that data. Actually, there are very few reasons for a name change after a user is registered

Reply:
I think it is expected to be able to correct any error or non-specific enough name. If the ninja is not mature enough for this basic responsibility, it should not even have an account and be only manage by the guardian. I guess if something unexpected happens, the organizer should block that ninja account.

Reply:
I really don't see why anyone should get to change their name in this case. In many platforms or services, like email and social media it makes sense, since it doesn't really matter in the grand scheme of things. But in my view we are trying to keep faithful records, and allowing for a name change without any warning to the organizers isn't really manageable. Isn't one of the reasons we have the manual account approval process so that we can approve the account and check if the information provided is accurate?

To me at least, the reasons users shouldn't be able to change their 'permanent' data here is the same reasons why UMinho's website doesn't allow you to do that.

Btw, I don't think *any* user should get to change that kind of 'permanent' data, like names and _maybe_ birthdays.

Reply:
Of course there are cases where a name change would be required, like a mistake that slipped through or a legal name change, but I don't see that being a problem often enough that it would warrant allowing everyone to do it on a whim.

Reply:
Well, If you decide basic information can't be changed, we just do it. And any of the cases you mention could be solved by requestion a organizer to fix it in the admin panel. I think I just followed the mock-up and it made sense to me that people could change their names as most regular platforms. I will merge like this but soon enough I should add social media fields and probably not allow name changes.